Assets/Scripts/PlayerController/PlayerMovement.cs

```diff
@@ -28,6 +28,9 @@ public class PlayerMovement : MonoBehaviour
     private float m_JumpHeight = 1.3f;
 
     private CharacterController m_CharacterController;
+    private TextChanger m_textChanger;
+    private PowerUpsManager m_PowerUps; 
```

Comment:
PowerUpsManager sounds important. Consider making it a Singleton: https://wiki.unity3d.com/index.php/Singleton
I believe Caleb has already committed the Singleton script into `e3`, so you can make `PowerupsManager` inherit `Singleton`.

Reply:
Sick stuff, kk will fetch then make the changes

Reply:
Actually on second thoughts, I am unable to make it a singleton for now as it requires different boolean values on the state of the power ups of each player. 

As each player requires their own instance of PowerUps Manager. As a result, we would need minimally 4. For now, I will not put it as a Singleton.